systemtest/src/test/java/io/strimzi/systemtest/upgrade/StrimziUpgradeST.java

```diff
@@ -205,7 +215,32 @@ void testUpgradeKafkaWithoutVersion() throws IOException {
         DeploymentUtils.waitTillDepHasRolled(KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME), 1, eoSnapshot);
 
         assertThat(kubeClient().getStatefulSet(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).getSpec().getTemplate().getSpec().getContainers()
-                .stream().filter(c -> c.getName().equals("kafka")).findFirst().get().getImage(), containsString("2.6.0"));
+                .stream().filter(c -> c.getName().equals("kafka")).findFirst().get().getImage(), containsString(latestKafkaVersion));
+    }
+
+    String getValueForLastKafkaVersionInFile(String field) throws IOException {
+        YAMLMapper mapper = new YAMLMapper();
+        JsonNode node = mapper.readTree(kafkaVersions);
+        ObjectNode kafkaVersionNode = (ObjectNode) node.get(node.size() - 1);
+
+        return kafkaVersionNode.get(field).asText();
+    }
+
+    void applyKafkaWithoutVersion(File kafka, String clusterName) throws IOException {
```

Comment:
Why not read the file into the `Kafka` class which you can configure directly? It would be easier than going through the JSON tree, or?